src/watcher.ts

```diff
@@ -31,12 +36,29 @@ export class Watcher extends events.EventEmitter {
   }
 
   start() {
-    // TODO: Clear based on dependency graph and file type.
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    fs.watch(this.pod.root, {recursive: true}, (event, filename) => {
-      // const podPath = `/${filename}`;
-      this.pod.cache.reset();
+    const reloadRegex = new RegExp(AUTORELOAD_PATHS.join('|'));
+    let timer = 0;
+    fs.watch(this.pod.root, {recursive: true}, (_event, filename) => {
+      // `filename` is either a numeric file descriptor or filename. Short
+      // circuit if this is a file descriptor.
+      const podPath = `/${filename}`;
+      if (!this.pod.fileExists(podPath)) {
+        return;
+      }
+      // `fs.watch` may trigger many times within a small window.
+      // Only run once within a 300ms window.
+      if (Date.now() - timer < 300) {
+        return;
+      }
+      if (podPath.match(reloadRegex)) {
+        this.server.reload();
```

Comment:
I find it weird that the watcher handles the server reload. In general you want to think about what the class is responsible for and follow the unix philosophy of creating small reusable classes. Otherwise here we've entangled file watching with server functions, and that creates a bit of spaghetti.

So I'd suggest having this call some of callback function whenever the files change, and having the server handle its own reload. For the console.log, the caller creating the server should be making that log statement.

Reply:
Per discussion, I refactored this to use event emitters more broadly and moved the callbacks on file watch change to the server itself. This had the downside of making the code less consolidated, and making the server itself have more code within it related to file watching callbacks, but it might be more idiomatic now.